gradle-plugin/src/main/kotlin/io/spine/protodata/gradle/plugin/Extension.kt

```diff
@@ -81,12 +83,64 @@ public class Extension(internal val project: Project): CodegenSettings {
     internal fun requestFile(forSourceSet: SourceSet): Provider<RegularFile> =
         requestFilesDirProperty.file(CodeGeneratorRequestFile.name(forSourceSet))
 
+    override val paths: Multimap<String, SourcePaths> = HashMultimap.create()
+
+    /**
+     * Obtains the source configured paths.
+     *
+     * If the deprecated [subDirs] and [targetBaseDir] are used, constructs [SourcePaths] instances
+     * from the present data for backward compatibility. However, in this compatibility mode,
+     * only Java and Kotlin source file sets can be constructed.
+     */
+    internal fun pathsOrCompat(sourceSet: SourceSet): Set<SourcePaths> {
```

Comment:
The problem here is that we need to remember which version (this one or `paths`) to use. And there is no clear sign on when to stop using one and start using another.

Maybe, we could have a "compatibility" mode turned on always. And for testing (or whatever we need "strictly new API" mode for) have another method, which would ignore the deprecated properties?

Reply:
I've added a note on this in doc. Essentially, this needs to exist until `mc-java` is updated. Potentially, until the very next PR. Then, only `paths` need to be used.